include/electionguard/precompute_buffers.hpp

```diff
@@ -0,0 +1,286 @@
+      {
+          unique_ptr<ElementModQ> new_rho = make_unique<ElementModQ>(*rho);
+          return move(new_rho);
+      }
+
+      unique_ptr<ElementModP> get_g_to_rho()
+      {
+          unique_ptr<ElementModP> new_g_to_rho = make_unique<ElementModP>(*g_to_rho);
+          return move(new_g_to_rho);
+      }
+
+      unique_ptr<ElementModP> get_pubkey_to_rho()
+      {
+          unique_ptr<ElementModP> new_pubkey_to_rho = make_unique<ElementModP>(*pubkey_to_rho);
+          return move(new_pubkey_to_rho);
+      }
+
+      void set(unique_ptr<ElementModQ> in_rho,
+               unique_ptr<ElementModP> in_g_to_rho,
+               unique_ptr<ElementModP> in_pubkey_to_rho);
+    protected:
+        void generateTriple(const ElementModP &publicKey);
+    };
+    
+    class EG_INTERNAL_API Quadruple
+    {
+        unique_ptr<ElementModQ> sigma;
+        unique_ptr<ElementModQ> rho;
+        unique_ptr<ElementModP> g_to_rho;
+        unique_ptr<ElementModP> g_to_sigma_mult_by_pubkey_to_rho;
+
+      public:
+        explicit Quadruple(const ElementModP &publicKey)
+        {
+            generateQuadruple(publicKey);
+        }
+        Quadruple() {};
+
+        unique_ptr<ElementModQ> get_sigma()
+        {
+            unique_ptr<ElementModQ> new_sigma = make_unique<ElementModQ>(*sigma);
+            return move(new_sigma);
+        }
+
+        unique_ptr<ElementModQ> get_rho()
+        {
+            unique_ptr<ElementModQ> new_rho = make_unique<ElementModQ>(*rho);
+            return move(new_rho);
+        }
+
+        unique_ptr<ElementModP> get_g_to_rho()
+        {
+            unique_ptr<ElementModP> new_g_to_rho = make_unique<ElementModP>(*g_to_rho);
+            return move(new_g_to_rho);
+        }
+
+        unique_ptr<ElementModP> get_g_to_sigma_mult_by_pubkey_to_rho()
+        {
+            unique_ptr<ElementModP> new_g_to_sigma_mult_by_pubkey_to_rho =
+              make_unique<ElementModP>(*g_to_sigma_mult_by_pubkey_to_rho);
+            return move(new_g_to_sigma_mult_by_pubkey_to_rho);
+        }
+
+        void set(unique_ptr<ElementModQ> in_rho, unique_ptr<ElementModQ> in_sigma,
+                 unique_ptr<ElementModP> in_g_to_rho,
+                 unique_ptr<ElementModP> in_g_to_sigma_mult_by_pubkey_to_rho);
+
+      protected:
+        void generateQuadruple(const ElementModP &publicKey);
+    };
+
+    class EG_INTERNAL_API TripleEntry
+    {
+      public:
+        explicit TripleEntry() {}
+
+        uint64_t *get_rho() { return rho; }
+        uint64_t *get_g_to_rho() { return g_to_rho; }
+        uint64_t *get_pubkey_to_rho() { return pubkey_to_rho; }
+
+        void set(unique_ptr<Triple> triple);
+        unique_ptr<Triple> get_triple();
+
+      private:
+        uint64_t rho[MAX_Q_LEN];
+        uint64_t g_to_rho[MAX_P_LEN];
+        uint64_t pubkey_to_rho[MAX_P_LEN];
+    };
+    
+    class EG_INTERNAL_API QuadrupleEntry
+    {
+      public:
+        explicit QuadrupleEntry() {}
```

Comment:
are we not following the rule of 5 anymore? i guess since it's an internal api it's probably not necessary but i noticed further down a lot of these objects follow a pattern where there is an explicit default constructor and then there's a set function. any particular reason to use that instead of just initializing the data in a parameterized constructor?

Reply:
Agreed we should make sure to follow the convention that we have in place.

Reply:
Removed all the set functions and added all constructors and operators for rule of 5.

Reply:
I have updated all objects to meet rule of 5 and got rid of the set methods.